src/core/net/dns_client.hpp

```diff
@@ -724,6 +806,33 @@ class Client : public InstanceLocator, private NonCopyable
 
 #endif // OPENTHREAD_CONFIG_DNS_CLIENT_SERVICE_DISCOVERY_ENABLE
 
+#if OPENTHREAD_CONFIG_DNS_CLIENT_ARBITRARY_RECORD_QUERY_ENABLE
+    /**
+     * Sends a DNS query for a given record type and name.
+     *
+     * The @p aConfig can be `nullptr`. In this case the default config (from `GetDefaultConfig()`) will be used as
+     * the config for this query. In a non-NULL @p aConfig, some of the fields can be left unspecified (value zero).
+     * The unspecified fields are then replaced by the values from the default config.
+     *
+     * @param[in] aRecordType   The resource record type to query.
+     * @param[in] aFirstLabel   The first label of the name to be queried (can be NULL if not needed).
+     * @param[in] aNextLabels   The next labels of the name to be queried (MUST NOT be NULL).
+     * @param[in] aCallback     A function pointer that shall be called on response reception or time-out.
+     * @param[in] aContext      A pointer to arbitrary context information used with @p aCallback.
+     * @param[in] aConfig       A pointer to the config to use for this query.
+     *
+     * @retval kErrorNone          Query sent successfully. @p aCallback will be invoked to report the outcome.
+     * @retval kErrorNotBufs       Insufficient buffer to prepare and send query.
```

Comment:
```suggestion
     * @retval kErrorNoBufs        Insufficient buffer to prepare and send query.
```

Reply:
Note there was one other instance of `kErrorNotBufs` in file dns_client.cpp  (existing code)

Reply:
Thanks. Both fixed.